tests/std/tests/P1032R1_Misc_constexpr/test.cpp

```diff
@@ -0,0 +1,195 @@
+        tuple<int, double> tuple_alloc_conversion_pair{allocator_arg_t{}, alloc, conversionInputPair};
+        tuple<int, double> tuple_alloc_conversion_const_pair{allocator_arg_t{}, alloc, constConversionInputPair};
+        tuple<int, double> tuple_alloc_copy{allocator_arg_t{}, alloc, tuple_alloc};
+        tuple<int, double> tuple_alloc_move{allocator_arg_t{}, alloc, std::move(tuple_alloc)};
+
+        swap(tuple_alloc, tuple_alloc_copy);
+        tuple_alloc.swap(tuple_alloc_copy);
+    }
+
+    // test array::swap
+    {
+        array<int, 2> array1{1, 2};
+        array<int, 2> array2{3, 4};
+
+        swap(array1, array2);
+        assert(array1[0] == 3 && array1[1] == 4 && array2[0] == 1 && array2[1] == 2);
+
+        array1.swap(array2);
+        assert(array1[0] == 1 && array1[1] == 2 && array2[0] == 3 && array2[1] == 4);
+
+        array<int, 0> array_empty1 = {};
+        array<int, 0> array_empty2 = {};
+
+        swap(array_empty1, array_empty2);
+        array_empty1.swap(array_empty2);
+    }
+
+    // test array::fill
+    {
+        array<int, 2> foo = {};
+        foo.fill(1);
+        assert(foo[0] == 1 && foo[1] == 1);
+    }
+
+    // test back_inserter
+    {
+        constexpr_container input;
+        int toBeMoved = 5;
+        auto tested   = back_inserter(input);
+
+        *tested++   = 42;
+        *(++tested) = 1729;
+        *tested++   = 1234;
+        tested      = 4;
+        tested      = std::move(toBeMoved);
+
+        assert(input.buffer[0] == 42 && input.buffer[1] == 1729 && input.buffer[2] == 1234 && input.buffer[3] == 4
+               && input.buffer[4] == 5 && input.buffer[5] == 0);
+    }
+
+    // test front_inserter
+    {
+        constexpr_container input;
+        int toBeMoved = 5;
+        auto tested   = front_inserter(input);
+
+        *tested++   = 42;
+        *(++tested) = 1729;
+        *tested++   = 1234;
+        tested      = 4;
+        tested      = std::move(toBeMoved);
+
+        assert(input.buffer[0] == 42 && input.buffer[1] == 1729 && input.buffer[2] == 1234 && input.buffer[3] == 4
+               && input.buffer[4] == 5 && input.buffer[5] == 0);
+    }
+
+    // test insert_inserter
+    {
+        constexpr_container input;
+        int toBeMoved = 5;
+        auto tested   = inserter(input, input.begin());
+
+        *tested++   = 42;
+        *(++tested) = 1729;
+        *tested++   = 1234;
+        tested      = 4;
+        tested      = std::move(toBeMoved);
+
+        assert(input.buffer[0] == 0 && input.buffer[1] == 42 && input.buffer[2] == 1729 && input.buffer[3] == 1234
+               && input.buffer[4] == 4 && input.buffer[5] == 5);
+    }
+
+    // test default_inserter
+    {
+        string_view in     = "This is some string";
+        string_view needle = "some";
+        default_searcher foo{needle.begin(), needle.end()};
+        auto [first, second] = foo(in.begin(), in.end());

```

Comment:
This feels like a bug. `first` and `second` are not used at all but there is no `unused variable` warning/error. which triggered everywhere else.

Could this be due to the implicit code generated by the compiler for the structured bindings?

Reply:
Perhaps it's because sometimes people want to ignore part of the output (I've heard they name variables `_` to do this), but yeah, seems like a diagnostic opportunity.